### PR TITLE
Set feature flags for opensource frontend

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:

Frontend PR for https://github.com/getsentry/sentry/pull/59956

### CODE REVIEW

File: static/app/views/settings/earlyFeatures/index.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  const access = new Set(organization.access);
+
+  return (
+    <Fragment>
+      <SentryDocumentTitle title={t('General Settings')} orgSlug={organization.slug} />
+      <div>
+        <SettingsPageHeader title={t('Early Features')} />
+        <PermissionAlert />
+
+        <EarlyFeaturesSettingsForm {...props} access={access} />
```
Comment (hubertdeng123) [inline static/app/views/settings/earlyFeatures/index.tsx:29]: Is this only showing up for self-hosted users?

Comment (markstory) [reply static/app/views/settings/earlyFeatures/index.tsx:29]: There is `settings.SENTRY_SELF_HOSTED` in the server side code.

File: static/app/views/settings/organization/navigationConfiguration.tsx [hunk 2]
```diff
@@ -90,6 +91,15 @@ const organizationNavigation: NavigationSection[] = [
         id: 'integrations',
         recordAnalytics: true,
       },
+      {
+        path: `${pathPrefix}/early-features/`,
+        title: t('Early Features'),
```
Comment (markstory) [inline static/app/views/settings/organization/navigationConfiguration.tsx:96]: Are sass customers supposed to see this as well?

File: static/app/views/settings/earlyFeatures/settingsForm.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+function EarlyFeaturesSettingsForm({organization, access, location}: Props) {
+  const {data: authProvider, isLoading: authProviderIsLoading} = useApiQuery<
+    Pick<State, 'authProvider'>
+  >([`/organizations/${organization.slug}/auth-provider/`], {
+    staleTime: 0,
+  });
+
+  const {data: featureFlags, isLoading: featurFlagsIsLoading} = useApiQuery<
```
Comment (hubertdeng123) [inline static/app/views/settings/earlyFeatures/settingsForm.tsx:33]: nit: `featureFlagIsLoading`

File: static/app/views/settings/earlyFeatures/settingsForm.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  const endpoint = `/internal/feature-flags/`;
+
+  if (authProviderIsLoading || featurFlagsIsLoading) {
+    return <LoadingIndicator />;
+  }
+
+  const initialData = Object.entries(featureFlags || {}).reduce(
+    (acc, [flag, obj]) => {
+      acc[flag] = obj.value;
```
Comment (hubertdeng123) [inline static/app/views/settings/earlyFeatures/settingsForm.tsx:47]: is this the right type?

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**